draft-irtf-cfrg-aegis-aead.md

```diff
@@ -1620,6 +1620,201 @@ Instead of relying on the generic `Encrypt` function, implementations can skip t
 
 After initialization, the `Update` function is called with constant parameters, allowing further optimizations.
 
+# AEGIS as a Message Authentication Code
+
+All AEGIS variants can be used to construct a MAC.
+
+For all the variants, The `Mac` function takes as input a key, a nonce, and data, and produces a 128- or 256-bit tag as output.
+
+This is the only function that permits the reuse of `(key, nonce)` pairs with different inputs.
+
+AEGIS-based MAC functions MUST NOT be used as hash functions. If the key is known, inputs causing state collisions can easily be crafted.
+Likewise, unlike hash-based MACs, tags MUST NOT be used for key derivation, as there is no guarantee they are uniformly random.
+
+~~~
+Mac(data, key, nonce)
+~~~
+
+Inputs:
+
+- `data`: the input data to be authenticate (length MUST be less than or equal to `A_MAX`).
+- `key`: the encryption key.
+- `nonce`: the public nonce.
+
+Outputs:
+
+- `tag`: the authentication tag.
+
+## AEGISMAC-128L
+
+AEGISMAC-128L refers to the `Mac` function built upon the building blocks of AEGIS-128L.
```

Comment:
Could we rephrase these bits to avoid saying `built` and `building`?

Reply:
Changed to "based on the building blocks", but I'm open to better ways to rephrase these bits!

Reply:
What do you think of just `based on AEGIS-128L`?

Reply:
That would work, too. I wrote "building blocks" to specifically refer to the function as they are defined in AEGIS-128L, rather than how the algorithm works in general.

Reply:
If you think "based on AEGIS-128L" is enough, I'm totally fine with that!

Reply:
Makes sense. Now that the tag computation has been changed, I think we should stick with what you've put.